audbackend/core/backend.py

```diff
@@ -637,13 +653,14 @@ def split(
 
         Raises:
             ValueError: if ``path`` contains invalid character
+                or does not start with ``'/'``
```

Comment:
Here it is similar to `join()`. There is no need that the path has to start with `'/'` as we don't communicate with the backend.

At the moment the docstring reads:
> Split path on backend into root and basename.

I guess the better wording might be: "Split path on backend into sub-path and basename."

BTW, what is the expected outcome of
```python
split('/f.ext')
```
Is it
```
('/', 'f.ext')
```
or
```
('/f.ext')
```
?
Maybe add a few explicit tests to `tests/test_backend.py` as `split()` is only indirectly tested at the moment.

Reply:
As discussed in https://github.com/audeering/audbackend/pull/104#discussion_r1176073420 we stay with raising the error when not starting with `/`

Reply:
Very good point!

As we output a tuple, I think it should be `('/', 'f.ext')`, then `join(split('/f.ext'))` will also work.

And I think the sub-path should always end on `/`, i.e.:

```
backend.split('/sub/f.ext')
```
```
('/sub/', 'f.ext')
```

Reply:
I agree regarding always returning a tuple, this is also in line with `os.path.split`:
```python
>>> os.path.split('/f.ext')
('/', 'f.ext')
```
I'm not so sure about always ending the sub-path on `/` as it is redundant information and not the behavior of `os.path.split`:
```python
>>> os.path.split('/a/f.ext')
('/a', 'f.ext')
```

Reply:
I think it makes sense as with `ls()` we also require that a sub-path ends on `/`. And otherwise, the following would not work:

```python
>>> backend.ls(backend.split('/sub/f.ext')[0])
```

There is also a fundamental difference between the path on a file-system and the path on a backend. On a file-system we cannot have a folder and file with the same name in the same directory. But with a backend path we don't have this limitation, e.g. you can do:

```python
>>> backend.put_file(file, '/a', '1.0.0')
>>> backend.put_file(file, '/a/a', '1.0.0')
```

So I think it is important we always distinguish between `/a` and `/a/`.

Reply:
And maybe we should then also raise an error with:

```
>>> backend.join('/sub', 'f.ext')
```

Reply:
But maybe it's fine like this, since the output of the function will be a valid backend path.

Reply:
Yes, I would still allow `backend.join('/sub', 'f.ext')` as you also allow for `backend.join('//sub//', 'f.ext')`

Reply:
I added tests for `split()` and also extended the docstring example to:

![image](https://user-images.githubusercontent.com/10383417/234205267-907f76b6-3dca-431b-9ae1-534a3034975b.png)

Reply:
Great, thanks